### PR TITLE
colorscheme bugfix

### PR DESCRIPTION
**Pull request by lanzhenw**
## What changes are proposed in this pull request?

- implemented design changes: add an info icon to named colorscale that links to guidelines
- fixed the following bugs:

- [x] On global setting tab, toggling between name/list, and modify values, colorscale jumps back to use previous value. 

- [x] If `dataset.app_config.colorscheme` is `null`, colorscheme cannot be saved to DB successfully.

- [x] `setDatasetColorScheme` cannot update the linked record successfully.

## How is this patch tested? If it is not, please explain why.

tested locally

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/components/ColorModal/colorPalette/Colorscale.tsx [hunk 1]
```diff
@@ -217,7 +228,20 @@ const Colorscale: React.FC = () => {
           />
           {tab === "name" && (
             <div>
-              Use a named plotly colorscale:
+              <Guide>
+                Use a named colorscale
+                <a
+                  href="https://plotly.com/python/colorscales/"
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/colorPalette/Colorscale.tsx:234]: Maybe move to ./utils/links.ts?

**Review by benjaminpkane — APPROVED** [review]
LGTM!

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx [hunk 2]
```diff
@@ -100,7 +104,7 @@ const ColorFooter: React.FC = () => {
                     );
 
                   const record = store.get(fragment.id);
-                  record && config?.setLinkedRecord(record, "colorScheme");
+                  record && config!.setLinkedRecord(record, "colorScheme");
```
Comment (manivoxel51) [inline app/packages/core/src/components/ColorModal/ColorFooter.tsx:107]: does this mean we want to ignore the fact that `config` can be undefined? maybe we should return on line 97

File: app/packages/core/src/components/ColorModal/colorPalette/Colorscale.tsx [hunk 3]
```diff
@@ -233,7 +257,7 @@ const Colorscale: React.FC = () => {
           )}
           {tab === "list" && (
             <div>
-              Define a custom colorscale:
+              Define a custom colorscale (range between 0 and 1):
```
Comment (manivoxel51) [inline app/packages/core/src/components/ColorModal/colorPalette/Colorscale.tsx:260]: 👍

**Review by manivoxel51 — DISMISSED** [review]
Nice work! 🍨 

tested locally. left two inline comments, 

one UX bug (unrelated to this PR but we should probably address)

- The manual input in the color selector disappears after entering 3 characters and seems to autocomplete the input. Can we have it stay open for the whole string?

https://github.com/voxel51/fiftyone/assets/109545780/f6b691dc-20e8-4a5b-8416-22f96dfa6040

UX issue 

- the tooltip on `select an attribute to color` dropdown is blocking the selection option. Can we have the tooltip open on the top instead

nit feedback - we can discuss offline

- I believe having a multi select dropdown for the color by value input would be useful. I personally always forget if the value was capitalized or not and what label values are available.

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx (hunk 2 above)
Comment (lanzhenw) [reply app/packages/core/src/components/ColorModal/ColorFooter.tsx:102]: `config` cannot be undefined. I will add a return.

**Review by benjaminpkane — APPROVED** [review]

**Pull request merged**